### PR TITLE
fix: avoid crash when opening tooltip or dimension modal in cases where option metadata is missing

### PR DESCRIPTION
**Pull request by jenniferarnesen**
Applies frontend fixes for [TECH-1056](https://dhis2.atlassian.net/browse/TECH-1056)

---

### Key features

1. For tooltip, if at least one of the option/legend names is not available, then don't add any html to the tooltip
image

3. For the dimension modal, if the name is not available, then use the code (i.e., Z4_VACCINE is the code in the screenshot below).
image

These fixes are workarounds and a proper backend fix should be coming in v2.39

### CODE REVIEW

**Comment by jenniferarnesen** [comment]
Maybe we can work with this after all.

File: src/components/Layout/TooltipContent.js [hunk 1]
```diff
@@ -85,7 +85,10 @@ export const TooltipContent = ({
         return itemDisplayNames
     }
 
-    const renderItems = (itemDisplayNames) => {
+    const renderItems = (itemDisplayNames = []) => {
+        if (itemDisplayNames.findIndex((name) => name === undefined) !== -1) {
```
Comment (martinkrulltott) [inline src/components/Layout/TooltipContent.js:89]:
```suggestion
        if (!itemDisplayNames.find((name) => name === undefined)) {
```

Comment (martinkrulltott) [reply src/components/Layout/TooltipContent.js:89]: A bit hard to follow what you're trying to do here though... do you want to make sure that all items to have a `name`?

Comment (martinkrulltott) [reply src/components/Layout/TooltipContent.js:89]: if so then I think this is more clear
```suggestion
        if (itemDisplayNames.some((name) => name === undefined)) {
```

Comment (martinkrulltott) [reply src/components/Layout/TooltipContent.js:89]: or even 
```suggestion
        if (itemDisplayNames.some((name) => !name)) {
```

Comment (jenniferarnesen) [reply src/components/Layout/TooltipContent.js:89]: Yeah that looks better

File: src/components/Layout/TooltipContent.js [hunk 2]
```diff
@@ -85,7 +85,10 @@ export const TooltipContent = ({
         return itemDisplayNames
     }
 
-    const renderItems = (itemDisplayNames) => {
+    const renderItems = (itemDisplayNames = []) => {
+        if (itemDisplayNames.some((name) => name === undefined)) {
```
Comment (martinkrulltott) [inline src/components/Layout/TooltipContent.js:89]:
```suggestion
        if (itemDisplayNames.every((name) => name)) {
```

File: src/components/Layout/TooltipContent.js [hunk 3]
```diff
@@ -85,7 +85,10 @@ export const TooltipContent = ({
         return itemDisplayNames
     }
 
-    const renderItems = (itemDisplayNames) => {
+    const renderItems = (itemDisplayNames = []) => {
+        if (itemDisplayNames.every((name) => name)) {
```
Comment (martinkrulltott) [inline src/components/Layout/TooltipContent.js:89]:
```suggestion
        if (!itemDisplayNames.every((name) => name)) {
```

Comment (jenniferarnesen) [reply src/components/Layout/TooltipContent.js:89]: I actually don't want this because if even one of the items is missing the name, then we don't want to show any of them.

Comment (martinkrulltott) [reply src/components/Layout/TooltipContent.js:89]: yeah so "all need to have a name", right. So that means "some dont have name" or "not every have name"... now it says "every has name", which is the inverse of what you want, because I was a bit too fast on the keyboard and forgot the "!", whoopsi..

**Review by martinkrulltott — APPROVED** [review]
